packages/flame/lib/src/components/core/component.dart

```diff
@@ -748,10 +759,25 @@ class Component {
     }
   }
 
+  @internal
+  LifecycleEventStatus handleLifecycleEventAdd(Component parent) {
+    assert(!isMounted);
```

Comment:
I believe our current practice is to add assertion messages only in cases when it is a user-caused error. For example, adding a component to two parents at once. In this case the message would tell the user what they did wrong, and how they can fix it.

At the same time, we have a small number of "internal" assertions, which should never trigger as far as we know. If such assertion fails, it would indicate a bug in Flame, some execution flow that we didn't think of. In these cases having no error message helps, because it makes the user go on Discord and start asking questions, and this is how we'd know there is a bug.

This assert falls into the second category. It also serves as a kind of "documentation" stating that I expect this method to be called when the component is not mounted yet.

Reply:
Alright, I think in some cases we should also add messages to our internal asserts so that it becomes easier for the us to know what 2e have done wrong, but I think these two are so self explanatory that it is not needed.

Reply:
sure, removed

Reply:
I think you misunderstood me, I meant that the asserts are good and self explanatory, so no message is needed (but the asserts themselves can stay. :sweat_smile:

Reply:
Lol, I meant to comment this for line 1056; I guess I accidentally posted in a wrong thread :)